sympy/polys/rings.py

```diff
@@ -1257,28 +1270,28 @@ def drop(self, gen):
                     del K[i]
                     poly[tuple(K)] = v
                 else:
-                    raise ValueError("Cannot drop %s" % gen)
+                    raise ValueError(f"Cannot drop {gen}")
 
             return poly
 
-    def _drop_to_ground(self, gen):
+    def _drop_to_ground(self, gen: PolyElement[Er] | int | str | None) -> tuple[int, PolyRing[Er] | PolyRing[Es]]:
```

Comment:
It would be better for `drop_to_ground` to convert `gen` to an `int` before calling this method.